jenkins/printJarVersion.sh

```diff
@@ -24,12 +25,14 @@ function print_ver(){
     
     if [[ "$VERSION" == *"-SNAPSHOT" ]]; then
         PREFIX=${VERSION%-SNAPSHOT}
-        TIMESTAMP=`grep -oP '(?<=timestamp>)[^<]+' < $REPO/maven-metadata-$SERVER_ID.xml`
-        BUILD_NUM=`grep -oP '(?<=buildNumber>)[^<]+' < $REPO/maven-metadata-$SERVER_ID.xml`
-        echo $TAG=$PREFIX-$TIMESTAMP-$BUILD_NUM$SUFFIX
+        # List the latest SNAPSHOT jar file in the maven repo
+        echo $TAG=`ls -t $REPO/$PREFIX-[0-9]*$SUFFIX | head -1 | xargs basename`
     else
         echo $TAG=$VERSION$SUFFIX
     fi
 }
 
-print_ver $1 $2 $3 $4 $5
+# only run in Jenkins build
+if [ -n "$JENKINS_URL" ]; then
```

Comment:
Rather than skipping **all** properties for a dev build, how about we add this condition to the existing SNAPSHOT check?  Then we'll get properties either way, but we'll know it's a dev build because the snapshot won't be timestamped.